### DISCUSSION TITLE
Support for other authentication methods aside MTLS

### DISCUSSION THREAD

**Discussion by gomezga1** [NVIDIA/NVFlare#1443] [Q&A]
Hi all,

It's great that the tool relies on an internal PKI and MTLS for client/server auth, however I find the model not really flexible as there are not any other alternatives. In case you want to expose the app through a WAF or any other network element enforcing MTLS, then the application won't work. Do you know if there are any plans to support other types of authentication?

Best regards,

Alex.

**Comment by chesterxgchen** [comment]
Alex, thanks for raising the question. So far, our customers haven't ask for such feature.  I assuming by "WAF", you mean "application fire wall".  can you describe your use case that requires alternative authentication Mechanism ? If there are enough users request such feature, we will ask product manger to add this for consideration.

**Reply by YuanTingHsieh** [reply]
@gomezga1 ^

**Comment by gomezga1** [comment]
Hi! Sorry for my late reply. Yes, by WAF I mean web application firewall. We would like to directly rely on the MTLs capabilities already provided by our WAF, and that's not feasible if MTLS is already in place at the app level.

**Reply by chesterxgchen** [reply]
Thanks for the clarification, we will considering this, hopefully we will be able to address this in future releases.
